autogpt/llm_utils.py

```diff
@@ -81,20 +84,28 @@ def create_chat_completion(
     for attempt in range(num_retries):
         backoff = 2 ** (attempt + 2)
         try:
-            if CFG.use_azure:
-                response = openai.ChatCompletion.create(
-                    deployment_id=CFG.get_azure_deployment_id_for_model(model),
-                    model=model,
-                    messages=messages,
-                    temperature=temperature,
-                    max_tokens=max_tokens,
-                )
-            else:
-                response = openai.ChatCompletion.create(
-                    model=model,
-                    messages=messages,
-                    temperature=temperature,
-                    max_tokens=max_tokens,
+            if CFG.llm_provider == "openai":
+                if CFG.use_azure:
+                    response = openai.ChatCompletion.create(
+                        deployment_id=CFG.get_azure_deployment_id_for_model(model),
+                        model=model,
+                        messages=messages,
+                        temperature=temperature,
+                        max_tokens=max_tokens,
+                    )
+                else:
+                    response = openai.ChatCompletion.create(
+                        model=model,
+                        messages=messages,
+                        temperature=temperature,
+                        max_tokens=max_tokens,
+                    )
+            if CFG.llm_provider == "anthropic":
+                response = anthropic_client.completion(
+                    prompt=f"{anthropic.HUMAN_PROMPT} {messages} {anthropic.AI_PROMPT}",
+                    stop_sequences = [anthropic.HUMAN_PROMPT],
+                    model="claude-v1.3",
+                    max_tokens_to_sample=10000000000,
```

Comment:
I don't think it'll work the way you expect (since the more tokens you support in sampling, the more it trims your input context window, and this will presumably trim off your entire prompt). By default, Claude tends to produce short responses in the 500-ish range unless you really push it hard. I'm less clear with AutoGPT responses, but suspect something like 1-2k default will work well here.

Reply:
That makes sense, updated here https://github.com/anthony-sarkis/Auto-GPT/commit/550c9e0e4256102a70c733377289f5dee0a7e3fc